tests/phpunit/test-wc-stripe-payment-gateway.php

```diff
@@ -568,4 +568,61 @@ public function test_get_balance_transaction_id_from_charge() {
 
 		$this->assertEquals( null, $this->gateway->get_balance_transaction_id_from_charge( null ) );
 	}
+
+	/**
+	 * Tests for Card brand and last 4 digits are displayed correctly for subscription.
+	 *
+	 * @see WC_Stripe_Subscriptions_Trait::maybe_render_subscription_payment_method()
+	 */
+	public function test_render_subscription_payment_method() {
+		$mock_subscription = WC_Helper_Order::create_order(); // We can use an order as a subscription.
+		$mock_subscription->set_payment_method( 'stripe' );
+
+		$mock_subscription->update_meta_data( '_stripe_source_id', 'src_mock' );
+		$mock_subscription->update_meta_data( '_stripe_customer_id', 'cus_mock' );
+		$mock_subscription->save();
+
+		// This is the key the customer's payment methods are stored under in the transient.
+		$transient_key = WC_Stripe_Customer::PAYMENT_METHODS_TRANSIENT_KEY . 'cardcus_mock';
+
+		$mock_payment_method       = new stdClass();
+		$mock_payment_method->id   = 'src_mock';
+		$mock_payment_method->type = 'card';
+		$mock_payment_method->card = new stdClass();
+
+		// VISA ending in 4242
+		$mock_payment_method->card->brand = 'visa';
+		$mock_payment_method->card->last4 = '4242';
+
+		set_transient( $transient_key, [ $mock_payment_method ], DAY_IN_SECONDS );
+		$this->assertEquals( 'Via Visa card ending in 4242', $this->gateway->maybe_render_subscription_payment_method( 'N/A', $mock_subscription ) );
+
+		// MasterCard ending in 1234
+		$mock_payment_method->card->brand = 'mastercard';
+		$mock_payment_method->card->last4 = '1234';
+
+		set_transient( $transient_key, [ $mock_payment_method ], DAY_IN_SECONDS );
+		$this->assertEquals( 'Via MasterCard card ending in 1234', $this->gateway->maybe_render_subscription_payment_method( 'N/A', $mock_subscription ) );
```

Comment:
Noting that https://github.com/woocommerce/woocommerce/pull/52101 will break this. I suppose we just wait until the changes land and update this when it breaks?